modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/storage/MetagraphMessageStorage.scala

```diff
@@ -0,0 +1,44 @@
+package org.tessellation.node.shared.infrastructure.snapshot.storage
+
+import cats.effect.Ref
+import cats.syntax.contravariantSemigroupal._
+import cats.{Functor, Semigroupal}
+
+import org.tessellation.schema.metagraphMessage.{AdminFeeKeyMessage, StakingMessage}
+
+trait MetagraphMessageStorage[F[_]] {
+  def getAdminFeeKeyMessage: F[Option[AdminFeeKeyMessage]]
+  def setAdminFeeKeyMessage(maybe: Option[AdminFeeKeyMessage]): F[Unit]
+  def getStakingMessage: F[Option[StakingMessage]]
+  def setStakingMessage(maybe: Option[StakingMessage]): F[Unit]
+}
+
+object MetagraphMessageStorage {
+  def make[F[_]: Functor: Semigroupal: Ref.Make](
+    maybeAdminFeeKeyMessage: Option[AdminFeeKeyMessage],
+    maybeStakingMessage: Option[StakingMessage]
+  ): F[MetagraphMessageStorage[F]] =
+    (
+      Ref.of[F, Option[AdminFeeKeyMessage]](maybeAdminFeeKeyMessage),
+      Ref.of[F, Option[StakingMessage]](maybeStakingMessage)
+    ).mapN(make(_, _))
+
+  def make[F[_]: Functor: Semigroupal: Ref.Make](
+    adminFeeKeyR: Ref[F, Option[AdminFeeKeyMessage]],
+    stakingR: Ref[F, Option[StakingMessage]]
+  ): MetagraphMessageStorage[F] =
+    new MetagraphMessageStorage[F] {
+      def getAdminFeeKeyMessage: F[Option[AdminFeeKeyMessage]] =
+        adminFeeKeyR.get
+
+      def setAdminFeeKeyMessage(maybe: Option[AdminFeeKeyMessage]): F[Unit] =
```

Comment:
Why maybe? At this point we should either set it or not I guess. None may only result from not ever setting this value.